chainer/links/normalization/batch_normalization.py

```diff
@@ -68,35 +69,132 @@ class BatchNormalization(link.Link):
             option, numbers in the tuple must be being sorted in ascending
             order. For example, (0, 2) is OK, but (2, 0) is not.
 
+    .. admonition:: Example
+
+        >>> x = np.arange(12).reshape(4, 3).astype(np.float32) ** 2
+        >>> x
+        array([[  0.,   1.,   4.],
+               [  9.,  16.,  25.],
+               [ 36.,  49.,  64.],
+               [ 81., 100., 121.]], dtype=float32)
+        >>> bn = chainer.links.BatchNormalization(3)
+        >>> bn(x)
+        variable([[-1.        , -1.0664359 , -1.1117983 ],
+                  [-0.71428573, -0.6714596 , -0.6401263 ],
+                  [ 0.14285715,  0.19748813,  0.23583598],
+                  [ 1.5714287 ,  1.5404074 ,  1.5160885 ]])
+        >>> (x - x.mean(axis=0)) / np.sqrt(x.var(axis=0) + 2e-5)
+        array([[-1.        , -1.0664359 , -1.1117983 ],
+               [-0.71428573, -0.6714596 , -0.6401263 ],
+               [ 0.14285715,  0.19748813,  0.235836  ],
+               [ 1.5714285 ,  1.5404074 ,  1.5160886 ]], dtype=float32)
+
+        There are several ways to make a BatchNormalization link.
+        Consider an input of batched 10 images of 32x32 with 3 channels.
+
+        >>> x = np.random.randn(10, 3, 32, 32).astype(np.float32)
+
+        1. Give the parameter size:
+
+            To normalize for each channel, give the number of channels
+            to ``size``.
+
+            >>> bn = chainer.links.BatchNormalization(3)
+            >>> bn.avg_mean.shape
+            (3,)
+            >>> y = bn(x)
+            >>> y.shape
+            (10, 3, 32, 32)
+            >>> np.testing.assert_allclose(
+            ...     y.array.mean(axis=(0, 2, 3)), 0, atol=1e-6)
+            >>> np.testing.assert_allclose(
+            ...     y.array.var(axis=(0, 2, 3)),
+            ...     x.var(axis=(0, 2, 3)) / (x.var(axis=(0, 2, 3)) + 2e-5),
```

Comment:
I think `np.ones(3, dtype=np.float32)` is simpler.

Reply:
I noticed that BN has `beta` and `gamma`, so, the output `y` is whitened (mean=0, var=1) only when `beta` and `gamma` are initial values or they are removed by setting  `use_gamma` and `use_beta` options to `True` (the example comes under the first case). Therefore, users might misunderstand and think `y` is always whitened if we put these assertions. (sorry, I should have noticed that earlier).

Reply:
Alternative ideas are:
1. Set `use_gamma` and `use_beta` to `False`.
2. Write as `y` is whitened because `beta` and `gamma` are initial values.
In both case, I would suggest to write that `y` is not whitened in general.